pokemon-api/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/facade/BadgeFacade.java

```diff
@@ -0,0 +1,15 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.facade;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeCreateDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.BadgeDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.GymDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.TrainerDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.ChallengeStatus;
+
+public interface BadgeFacade {
+    Long createBadge(BadgeCreateDTO badge);
+    BadgeDTO findBadgeById(Long id);
+    GymDTO getGymOfBadge(Long id);
+    TrainerDTO getTrainerOfBadge(Long id);
+    void updateStatusOnBadge(Long id, ChallengeStatus status);
```

Comment:
I would break this out to several methods - issue, revoke. We also need to check that whoever requested the change is permitted to do so (i.e. that person is the leader of the gym of the badge) - so we need to pass that information somehow.

Reply:
So there should be two methods, one for changing the status to WIN and the other one which changes it to LOST? Do I understand it correctly?

Reply:
More or less - in total I can see 3 or 4 different methods: one for each transition - waiting -> won (challenger won), waiting -> lost (challenger lost), won -> revoked (breach of pkmn league rules), and maybe lost->waiting (when gym is rechallenged - only if business equality doesn't allow creating new badge for same gym - challenger pair)